CONVENTIONS.md

```diff
@@ -85,3 +85,86 @@ metal3-io](https://github.com/metal3-io/metal3-docs/blob/master/design/bare-meta
 ### API
 
 OpenShift APIs follow the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md).
+
+### Operators
+
+The OpenShift project is an early adopter of, and makes extensive use of, [the
+operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/),
+and so it is incumbent on us to establish some conventions around operators.
+
+#### Taints and Tolerations
+
+An operator deployed by the CVO should run on master nodes and therefore should
+tolerate the following taint:
+
+* `node-role.kubernetes.io/master`
+
+For example:
+
+```yaml
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+```
+
+Tolerating this taint should suffice for the vast majority of core OpenShift
+operators.  In exceptional cases, an operator may tolerate one or more of the
+following taints if doing so is necessary to form a functional Kubernetes node:
+
+* `node.kubernetes.io/disk-pressure`
+* `node.kubernetes.io/memory-pressure`
+* `node.kubernetes.io/network-unavailable`
+* `node.kubernetes.io/not-ready`
+* `node.kubernetes.io/pid-pressure`
+* `node.kubernetes.io/unreachable`
+
+Operators should not specify tolerations in their manifests for any of the taints in
+the above list without an explicit and credible justification.
+
+When an operator configures its operand, the operator likewise may specify
+tolerations for the aforementioned taints but should do so only as necessary and only
+with explicit justification.
+
+Note that the DefaultTolerationSeconds and PodTolerationRestriction admission plugins
+may add time-bound tolerations to an operator or operand in addition to the
```

Comment:
I remember @sjenning getting worked up over this, but I have no memory of what the details were